syft/cataloging/license.go

```diff
@@ -13,3 +13,15 @@ func DefaultLicenseConfig() LicenseConfig {
 		Coverage:                    licenses.DefaultCoverageThreshold,
 	}
 }
+
+// GetContextKey allows callers to access the key used by
+// syft to store the license scanner in the context
+func GetContextKey() licenses.LicenseScannerKey {
+	return licenses.CtxKey
+}
+
+// NewDefaultLicenseScanner creates a default license scanner and exists
+// to export NewDefaultScanner to those who use Syft as a library.
+func NewDefaultLicenseScanner(o ...licenses.Option) (licenses.Scanner, error) {
```

Comment:
A user would not be able to use this due to the internal options. Also, it would not be ideal to return the internal `licenses.Scanner`. However, I would like to avoid exporting anything we don't need to, as it will become part of the Syft 1.x API henceforth.